### PR TITLE
Add new WebView2 tests, add focus logging to WebView2 tests, avoid potential flicker when removing

### PR DESCRIPTION
**Pull request by krschau**
Add a check in CheckAndUpdateWebViewPosition() to not do work when the WebVIew2 is not loaded, to prevent potential visual flicker.

Adds two new WebView2 tests: CreateWebView2OffTreeTest and HtmlDropdownTest. Splits NavigationStartingTest into two tests, NavigationStartingTest and NavigationStartingInvalidTest because as one long test they can sometimes timeout.

Adds logging for the focus and blur events in some of the test web pages.

### CODE REVIEW

File: dev/WebView2/InteractionTests/WebView2Tests.cs [hunk 1]
```diff
@@ -683,11 +640,8 @@ public void NavigationErrorTest()
         // For each of the Focus_* tests: 
         // 1) We Tab/Shift+Tab/Click amongst xaml controls(x1, x2) and web controls(w1, w2). 
         // 2) After each such change:
-        //     a. Validate Edge focus state by hovering over UpdateAnaheimFocusTextBlock, which triggers a tooltip handler 
-        //        in the (Xaml) test app, which tests the color of a rectangle in the webpage that reflects the focused web control(w1, w2, or n/a).
-        //     b. Validate Xaml focus using MITA API's directly. Since WebView2 doesn't have accessibility hooked up yet,
-        //       the FocusAcquiredWaiter will not work correctly with it. Such validation is temporarily commented out 
-        //       with the TODO_WebView2_Accessibility tag in the comments.
+        //     a. Validate Edge focus state by waiting for web message sent when an element gets focu
```
Comment (DmitriyKomin) [inline dev/WebView2/InteractionTests/WebView2Tests.cs:643]: >lement gets focu [](http://example.com/codeflow?start=85&length=16)

spelling

File: dev/WebView2/InteractionTests/WebView2Tests.cs [hunk 2]
```diff
@@ -2179,6 +2077,78 @@ public void NonAsciiUriTest()
             }
         }
 
+        [TestMethod]
+        [TestProperty("TestSuite", "D")]
+        public void CreateWebView2OffTreeTest()
```
Comment (DmitriyKomin) [inline dev/WebView2/InteractionTests/WebView2Tests.cs:2082]: > public void CreateWebView2OffTreeTest() [](http://example.com/codeflow?start=7&length=40)

This basic scenario is covered in BasicCoreObjectCreationAndDestructionTest (specifically Preload Test 1 & 2 subtests). But here you also valiate input, which I agree is very valuable. Maybe change the name to emphasize that (eg "OffTreeWebViewInputTest")? If that's too much trouble, just adding a comment about this here is good too.

File: dev/WebView2/InteractionTests/WebView2Tests.cs [hunk 3]
```diff
@@ -328,15 +327,6 @@ private static void CompleteTestAndWaitForResult(string selectedTest)
...
 
-            for (int i = 0; i <= 2; i++)
-            {
-                if (expectedValue != result.Value)
-                {
-                    Log.Comment($"CompleteTestAndWaitForResult: Expected '{expectedValue}', result was '{result.Value}'. Waiting...");
-                    Thread.Sleep(TimeSpan.FromMilliseconds(500));
-                }
-            }
-
```
Comment (DmitriyKomin) [inline dev/WebView2/InteractionTests/WebView2Tests.cs:339]: Great to see this gone, wondering why ever needed it (and can remove now)?

File: dev/WebView2/InteractionTests/WebView2Tests.cs [hunk 4]
```diff
@@ -702,47 +656,31 @@ public void Focus_BasicTabTest()
...
-                // TODO_WebView2_Accessibility: Verify.IsTrue(webviewControl.HasKeyboardFocus);
-
-                // TODO_WebView2_Accessibility: using (var xamlFocusWaiter = new FocusAcquiredWaiter(UICondition.CreateFromId("x2")))=
                 Log.Comment("Tab w2 -> x2");
-                KeyboardHelper.PressKey(Key.Tab);
-                // TODO_WebView2_Accessibility : xamlFocusWaiter.Wait();
-                WaitForFocus("n/a");
-
-                Verify.IsTrue(x2.HasKeyboardFocus);
+                using (var xamlFocusWaiter = new FocusAcquiredWaiter("TabStopButton2"))
```
Comment (DmitriyKomin) [inline dev/WebView2/InteractionTests/WebView2Tests.cs:677]: >FocusAcquiredWaiter [](http://example.com/codeflow?start=49&length=19)

Great to see this!

File: dev/WebView2/TestUI/WebView2BasicPage.xaml [hunk 5]
```diff
@@ -59,6 +61,7 @@
                             <ComboBoxItem AutomationProperties.Name="NavigateToVideoTest">NavigateToVideoTest</ComboBoxItem>
                             <ComboBoxItem AutomationProperties.Name="NavigationErrorTest">NavigationErrorTest</ComboBoxItem>
                             <ComboBoxItem AutomationProperties.Name="NavigationStartingTest">NavigationStartingTest</ComboBoxItem>
+                            <ComboBoxItem AutomationProperties.Name="NavigationStartingInvalidTest">NavigationStartingInvalidTest</ComboBoxItem>
```
Comment (DmitriyKomin) [inline dev/WebView2/TestUI/WebView2BasicPage.xaml:64]: Nit: should go before "NavigationStartingTest"

**Comment by DmitriyKomin** [comment]
> <?xml version="1.0" encoding="utf-8"?>

Any changes in this file?

---
Refers to: test/MUXControlsTestApp/MSTest/MUXControlsTestApp.csproj:1 in 65f8bc6. [](commit_id = 65f8bc6bac997c1f077488b0a9e46d1483309a0a, deletion_comment = False)

**Review by DmitriyKomin — CHANGES_REQUESTED** [review]
:clock1:

**Comment by krschau** [comment]
> > 
> 
> Any changes in this file?
> 
> Refers to: test/MUXControlsTestApp/MSTest/MUXControlsTestApp.csproj:1 in [65f8bc6](https://github.com/microsoft/microsoft-ui-xaml/commit/65f8bc6bac997c1f077488b0a9e46d1483309a0a). [](commit_id = [65f8bc6](https://github.com/microsoft/microsoft-ui-xaml/commit/65f8bc6bac997c1f077488b0a9e46d1483309a0a), deletion_comment = False)

There was a stray tab I changed to spaces.

**Review by DmitriyKomin — APPROVED** [review]
:shipit:

**Pull request merged**